articles/machines/resources-estimator.md

```diff
@@ -0,0 +1,154 @@
+            MyQuantumProgram.Run(estimator).Wait();
+            Console.WriteLine(estimator.ToTSV());
+        }
+    }
+}
+```
+
+As the example shows, the `ResourcesEstimator` provides a `ToTSV()` method to generate
+a table with tab-seperated-values (TSV) that can be saved into a file
+or written to the console for analysis. The output of the above program should look something like this:
+
+```Output
+Metric          Sum
+CNOT            1000
+QubitClifford   1000
+R               0
+Measure         4002
+T               0
+Depth           0
+Width           2
+BorrowedWidth   0
+```
+
+> [!NOTE]
+> The `ResourcesEstimator` does not reset its calculations on every run, if the same instance is
+> used to execute another operation it will keep aggregating counts on top of existing results.
+> If you need to reset calculations across runs, create a new instance for every execution.
+
+
+## Programmatically Retrieving the Estimated Data
+
+In addition to a TSV table, the resources estimated can be retrieved programmatically
+via the `ResourcesEstimator`'s `Data` property. `Data` provides a `System.DataTable` 
+instance with two columns: `Metric` and `Sum`, indexed by the metrics names.
+
+The following code shows how to retrieve and print the total number of `QubitClifford`, `T` and `CNOT` 
+gates used by a Q# operation:
+
+```csharp
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.Simulators;
+
+namespace Quantum.MyProgram
+{
+    class Driver
+    {
+        static void Main(string[] args)
+        {
+            ResourcesEstimator estimator = new ResourcesEstimator();
+            var res = MyQuantumProgram.Run(estimator).Result;
+
+            var data = estimator.Data;
+            Console.WriteLine($"QubitCliffords: {data.Rows.Find("QubitClifford")["Sum"]}");
+            Console.WriteLine($"Ts: {data.Rows.Find("T")["Sum"]}");
+            Console.WriteLine($"CNOTs: {data.Rows.Find("CNOT")["Sum"]}");
+        }
+    }
+}
+```
+
+## Metrics Reported
+
+The following is the list of metrics estimated by the `ResourcesEstimator`:
+
+* __CNOT__: The count of CNOT (also known as the Controlled Pauli X gate) gates executed.
+* __QubitClifford__: The count of any single qubit Clifford and Pauli gates executed.
+* __Measure__:  The count of any measurements executed.
+* __R__: The count of any single qubit rotations executed, excluding T, Clifford and Pauli gates.
+* __T__: The count of T gates and their conjugates, including the T gate, T_x = H.T.H, and T_y = Hy.T.Hy, executed.
+* __Depth__: Depth of the quantum circuit executed by the Q# operation.
+* __Width__: Maximum number of qubits allocated during the execution of the Q# operation.
+* __BorrowedWidth__: Maximum number of qubits borrowed inside the Q# operation.
+
+
+## Providing the Probability of Measurement Outcomes
+
+<xref:microsoft.quantum.primitive.assertprob> from the <xref:microsoft.quantum.primitive> namespace can 
```

Comment:
These references do not display properly.

Reply:
It's because they point to https://review.docs.microsoft.com/, they work fine in docs.microsoft.com, which is what will be generated once in `live`